wadm/lib/wadm/model/store.ex

```diff
@@ -0,0 +1,222 @@
+                   ["SADD", lattice_set_key, vetted_model.name]
+                 ]
+               ) do
+          :ok
+        else
+          e ->
+            {:error, "Failed to add model version to Redis: #{inspect(e)}"}
+        end
+      end
+    else
+      {:error, e} ->
+        {:error, e}
+    end
+  end
+
+  @spec delete_model_version(Redix.connection(), String.t(), String.t(), String.t()) ::
+          :ok | {:error, String.t()}
+  @doc """
+  This function removes a model version from the persistent store.
+  """
+  def delete_model_version(redis, model_name, version, lattice_id)
+      when is_binary(model_name) and is_binary(version) and is_binary(lattice_id) do
+    set_key = "wadm:#{lattice_id}:#{model_name}:versions"
+    detail_key = "wadm:#{lattice_id}:#{model_name}:versions:#{version}"
+
+    with {:ok, [1, 1]} <-
+           Redix.transaction_pipeline(
+             redis,
+             [
+               ["ZREM", set_key, version],
+               ["DEL", detail_key]
+             ]
+           ) do
+      :ok
+    else
+      {:ok, [0, 0]} ->
+        Logger.warn("Attempt to remove a non-existent model version from the store")
+        :ok
+
+      e ->
+        {:error, "Failed to delete model version from Redis: #{inspect(e)}"}
+    end
+  end
+
+  @spec get_model_version(Redix.connection(), String.t(), String.t(), String.t()) ::
+          {:ok, ModelMap.t()} | {:error, String.t()}
+  @doc """
+  This function retrieves the details for a specific model version. This includes
+  both the originally submitted raw text and the post-validation, normalized model
+  """
+  def get_model_version(redis, model_name, version, lattice_id)
+      when is_binary(model_name) and is_binary(version) and is_binary(lattice_id) do
+    detail_key = "wadm:#{lattice_id}:#{model_name}:versions:#{version}"
+
+    with {:ok, raw} <- Redix.command(redis, ["GET", detail_key]) do
+      if raw != nil do
+        {:ok, raw |> :erlang.binary_to_term()}
+      else
+        {:error, "No such model version in store: #{model_name} #{version}"}
+      end
+    else
+      e ->
+        {:error, "Failed to retrieve model version: #{inspect(e)}"}
+    end
+  end
+
+  @spec model_version_exists?(Redix.connection(), String.t(), String.t(), String.t()) :: boolean()
+  @doc """
+  Perform a safe check to see if a model version exists. Consumers should call this to verify
+  the existence of a model version before performing other actions that might fail on non-existent
+  version details
+  """
+  def model_version_exists?(redis, model_name, version, lattice_id) do
+    detail_key = "wadm:#{lattice_id}:#{model_name}:versions:#{version}"
+
+    with {:ok, 1} <- Redix.command(redis, ["EXISTS", detail_key]) do
+      true
+    else
+      _ ->
+        false
+    end
+  end
+
+  defp stamp_to_rfc_3339(stamp) when is_integer(stamp) do
+    with {:ok, dt} <- DateTime.from_unix(stamp, :millisecond) do
+      dt |> DateTime.to_iso8601()
```

Comment:
So this goes to iso8601 and not rfc3339. They are basically the same, but as far as I understand the `T` separator is the main difference. I don't care which one we use, but we should make sure it is the right format